File: Cpp/06_data_structures.cpp
#include <iostream>
#include <vector>
#include <map>
#include <set>

int main() {
    // Vector (dynamic array)
    std::vector<int> nums = {1, 2, 3, 4, 5};
    nums.push_back(6);

    // Map (dictionary)
    std::map<std::string, int> person = {{"age", 28}, {"city", 0}}; // demo

    // Set
    std::set<int> unique = {1, 2, 2, 3};

    std::cout << "Vector size: " << nums.size() << std::endl;
    std::cout << "Unique set size: " << unique.size() << std::endl;
    return 0;
}

File: Java/06_data_structures.java
import java.util.*;

public class DataStructures {
    public static void main(String[] args) {
        // ArrayList (dynamic array)
        List<Integer> nums = new ArrayList<>(Arrays.asList(1, 2, 3, 4, 5));
        nums.add(6);

        // HashMap (dictionary)
        Map<String, Integer> person = new HashMap<>();
        person.put("age", 28);

        // HashSet
        Set<Integer> unique = new HashSet<>(Arrays.asList(1, 2, 2, 3));

        System.out.println("List size: " + nums.size());
        System.out.println("Unique set size: " + unique.size());
    }
}
